akka-stream/src/main/scala/akka/stream/impl/ActorRefSinkActor.scala

```diff
@@ -0,0 +1,36 @@
+package akka.stream.impl
+
+import akka.stream.actor.ActorSubscriber
+import akka.actor.ActorRef
+import akka.stream.actor.ActorSubscriberMessage
+import akka.actor.Status
+import akka.stream.actor.WatermarkRequestStrategy
+import akka.actor.Props
+import akka.actor.Terminated
+
+/**
+ * INTERNAL API
+ */
+private[akka] object ActorRefSinkActor {
+  def props(ref: ActorRef, highWatermark: Int, onCompleteMessage: Any): Props =
+    Props(new ActorRefSinkActor(ref, highWatermark, onCompleteMessage))
+}
+
+/**
+ * INTERNAL API
+ */
+private[akka] class ActorRefSinkActor(ref: ActorRef, highWatermark: Int, onCompleteMessage: Any) extends ActorSubscriber {
```

Comment:
it would be interesting to experiment later if this can be implemented via a lightweight synthetic actorRef just like PromiseActorRef.